### ISSUE TITLE
Bug: `WorkspaceItem.getContainer()` always returns the root container

### ISSUE THREAD

**Issue by ofalvai** [obsidianmd/obsidian-api#160]
I'm not sure if this is a bug or the intended behavior, but here is what I want to achieve:

My plugin opens a custom view by `workspace.getLeaf("split", "vertical")` (this is the behavior of the built-in "Open to the right" context menu if I'm not mistaken). But if there is an existing vertical split in the right split, I don't want to add a third pane, but open a tab in the existing split. So I came up with this code that looks up the existing instance of my custom view, use its parent to open a new tab:

```ts
const chatViewLeaves = workspace.getLeavesOfType(VIEW_TYPE_NOTE_CHAT)
if (chatViewLeaves.length > 0) {
  const containerOfChatViews = chatViewLeaves[0].getContainer()
  leaf = workspace.createLeafInParent(containerOfChatViews, 1)
  // ...rest of the view creation logic
}
```

I was expecting this to open a leaf in the `tabs` split, but the new view became another leaf in the root split (essentially another vertical pane). Also, in this workspace setup, `chatViewLeaves[0].getContainer()` and `chatViewLeaves[0].getRoot()` return the same reference.

Is it possible what I'm trying to achieve? Or is it an anti-pattern and I should not even try this? Any help would be appreaciated.

**Comment by liamcain** [comment]
`getContainer` is working as intended, it is meant to help distinguish between tabs that are inside a sidebar from tabs in the main area of the window.

Instead of `getContainer()` what you actually want to access is `leaf.parent` which doesn't appear to be exposed in the API. To find the parent WorkspaceSplit, you need to traverse up the workspace tree and find the closest split.

```ts
function getParentSplit(item: WorkspaceLeaf) {
  let ancestor = item.parent;
  while (ancestor && !(ancestor instanceof WorkspaceSplit)) {
    ancestor = ancestor.parent;
  }
  return ancestor;
}

const chatViewLeaves = workspace.getLeavesOfType(VIEW_TYPE_NOTE_CHAT)
if (chatViewLeaves.length > 0) {
  const split = getParentSplit(chatViewLeaves[0]);
  leaf = workspace.createLeafInParent(split, 1);
  // ...rest of the view creation logic
}
```

Note: this code will throw errors in your editor, so you'll need to use `@ts-ignore` for now

**Comment by ofalvai** [comment]
Thank you @liamcain, I tried the suggested snippet, and even though the returned split instance is correct (I double-checked the DOM reference in devtools), the end result is still the same: my custom view is directly attached to the root split without any tab UI. Is it possible that `createLeafInParent()` can't create tab leaves and it is hardcoded to create a child of a `split`?

I double-checked the `workspace.json` structure on disk, and it matches what I see on the UI:

```json
{
  "main": {
    "id": "db68a861fda8a4a5",
    "type": "split",
    "children": [
      {
        "id": "07c59eaad420ce86",
        "type": "leaf",
        "state": {
          "type": "llm-note-chat-view",
          "state": {}
        }
      },
      {
        "id": "1b2c279856192e00",
        "type": "tabs",
        "children": [...]
      }
    ]
}
```

For the record, I also tried `workspace.duplicateLeaf(chatViewLeaves[0], "tab")` and it gives the same result.
